src/pages/index.js

```diff
@@ -0,0 +1,323 @@
+const postCaptionInput = postModalElement.querySelector("#caption");
+
+const previewModalElement = document.querySelector("#preview-modal");
+const previewImageElement = previewModalElement.querySelector(".modal__image");
+const previewCaptionElement =
+  previewModalElement.querySelector(".modal__caption");
+const previewCloseButtonElement = previewModalElement.querySelector(
+  ".modal__close-button"
+);
+
+const cardTemplate = document.querySelector("#card-template");
+const cardsListElement = document.querySelector(".cards__list");
+
+let selectedCard;
+let selectedCardId;
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModalElement);
+}
+
+function handleLike(evt, id) {
+  const likeButtonElement = evt.target;
+  const isLiked = likeButtonElement.classList.contains(
+    "card__like-button_liked"
+  );
+
+  api
+    .changeLikeStatus(id, isLiked)
+    .then((data) => {
+      likeButtonElement.classList.toggle("card__like-button_liked");
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameElement = cardElement.querySelector(".card__label");
+  const cardImageElement = cardElement.querySelector(".card__image");
+  const cardLikeButtonElement = cardElement.querySelector(".card__like-button");
+  const cardDeleteButtonElement = cardElement.querySelector(
+    ".card__delete-button"
+  );
+
+  cardNameElement.textContent = data.name;
+  cardImageElement.src = data.link;
+  cardImageElement.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeButtonElement.classList.add("card__like-button_liked");
+  }
+
+  cardLikeButtonElement.addEventListener("click", (evt) =>
+    handleLike(evt, data._id)
+  );
+
+  cardDeleteButtonElement.addEventListener("click", (evt) =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  cardImageElement.addEventListener("click", () => {
+    openModal(previewModalElement);
+    previewImageElement.src = data.link;
+    previewImageElement.alt = data.name;
+    previewCaptionElement.textContent = data.name;
+  });
+
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeOnEscape);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeOnEscape);
+}
+
+function setupCloseOnOverlayClick(modalList) {
+  modalList.forEach((modal) => {
+    modal.addEventListener("click", (evt) => {
+      if (evt.target === modal && modal.classList.contains("modal_opened")) {
+        closeModal(modal);
+      }
+    });
+  });
+}
+
+setupCloseOnOverlayClick(modalList);
+
+function closeOnEscape(evt) {
+  const openedModal = document.querySelector(".modal.modal_opened");
```

Comment:
You need to find the opened modal only when you have pressed `Escape`. So it should be done inside `if`-construction, otherwise you find it every time you type anything in the inputs

```js
function closeOnEscape(evt) {
  if (evt.key === "Escape" || evt.keyCode === "27") {
    const openedModal = document.querySelector(".modal.modal_opened");
    openedModal && closeModal(openedModal);
  }
}
```